mne/preprocessing/bads.py

```diff
@@ -38,3 +38,176 @@ def find_outliers(X, threshold=3.0, max_iter=2):
+    is repeated with the mean of the maps identified in the first stage.
+    
+    Outputs a list of fitted ICAs with the indices of the marked ICs in a 
+    specified field.
+    
+    The original Corrmap website: www.debener.de/corrmap/corrmapplugin1.html
+    
+    Parameters
+    ----------
+    icas : list
+        A list of fitted ICAs.
+    template : (int, int) tuple
+        Index of the list member from which the template should be obtained, and
+        the index of the IC.
+    threshold : "auto" | list of floats > 0 < 1 | float > 0 < 1 | float > 1
+        Correlation threshold for identifying ICs
+        If "auto": search for the best map by trying all correlations between 
+        0.6 and 0.95. In the original proposal, lower values are considered, but
+        this is not yet implemented.
+        If list of floats: search for the best map in the specified range of 
+        correlation strengths.
+        If float > 0: select ICs correlating better than this.
+        If float > 1: use find_outliers to identify ICs within subjects (not in 
+        original Corrmap)
+        Defaults to "auto".
+    name : string
+        Name of the field under which found ICs should be specified.
+        Defaults to "marked_ics".
+    verbose_return : bool
+        Should the output be restricted to the list of ICs (False), or also the 
+        correlation of the constructed template with the selected maps and the 
+        correlations?
+        Defaults to False.
+    plot : bool
+        Should constructed template and selected maps be plotted?
+    ----------
+    Returns : list 
+        Returns a list of fitted ICs, enrichened with the indices of the selected
+        maps in the field specified by the name keyword.   
+    """
+
+    from mne.preprocessing.bads import find_outliers
+    import numpy as np
+
+    def vcorrcoef(X,y): # vectorised correlation, by Jon Herman
+        Xm = np.reshape(np.mean(X,axis=1),(X.shape[0],1))
+        ym = np.mean(y)
+        r_num = np.sum((X-Xm)*(y-ym),axis=1)
+        r_den = np.sqrt(np.sum((X-Xm)**2,axis=1)*np.sum((y-ym)**2))
+        r = r_num/r_den
+        return r
+
+    def get_ica_map(ica, components=None):
+        if not components:
+            components = range(ica.n_components_)
+        maps = np.dot(ica.mixing_matrix_.T, ica.pca_components_)[components]
+        return maps
+
+    def find_max_corrs(all_maps, target, threshold):
+        all_corrs = [vcorrcoef(subj, target) for subj in all_maps]
+        abs_corrs = [np.abs(a) for a in all_corrs]
+        corr_polarities = [np.sign(a) for a in all_corrs]
+
+        if threshold < 1:
+            max_corrs = [list(np.nonzero(s_corr>threshold))
+                         for s_corr in abs_corrs]
+        else:     
+            max_corrs = [list(find_outliers(s_corr, threshold = threshold))
+                         for s_corr in abs_corrs]
+
+        am = []
+        [am.extend(list(a[m])) for a, m in zip(abs_corrs, max_corrs)]
```

Comment:
why create am in the first place and extend it? you could also just do the list comprehension, not?